jwst/saturation/tests/test_saturation.py

```diff
@@ -0,0 +1,344 @@
+    data.data[0, 2, 500, 500] = 40000
+    data.data[0, 3, 500, 500] = 60000   # Signal reaches saturation limit
+    data.data[0, 4, 500, 500] = 62000
+
+    # Get reference file
+    reffile = "/grp/crds/cache/references/jwst/jwst_nircam_saturation_0064.fits"
+
+    # Run the pipeline
+    output = SaturationStep.call(data, override_saturation=reffile)
+
+    # Check that correct pixel and group 3+ are flagged as saturated
+    # Check that other pixel and groups are not flagged
+    assert dqflags.group['SATURATED'] == np.max(output.groupdq[0, :, 500, 500])
+    assert np.all(output.groupdq[0, 3:, 500, 500] == dqflags.group['SATURATED'])
+    assert np.all(output.groupdq[0, :3, 500, 500] != dqflags.group['SATURATED'])
+    assert np.all(output.groupdq[0, :, 100, 100] != dqflags.group['SATURATED'])
+
+
+@pytest.fixture(scope='function')
+def setup_nrc_cube():
+    ''' Set up fake NIRCam data to test.'''
+
+    def _cube(ngroups, nrows, ncols):
+
+        nints = 1
+
+        data_model = RampModel()
+        data_model.data = np.zeros(shape=(nints, ngroups, nrows, ncols), dtype=np.float32)
+        data_model.pixeldq = np.zeros(shape=(nrows, ncols), dtype=np.int32)
+        data_model.groupdq = np.zeros(shape=(nints, ngroups, nrows, ncols), dtype=np.float32)
+        data_model.meta.subarray.xstart = 1
+        data_model.meta.subarray.ystart = 1
+        data_model.meta.subarray.xsize = ncols
+        data_model.meta.subarray.ysize = nrows
+        data_model.meta.exposure.ngroups = ngroups
+        data_model.meta.instrument.name = 'NIRCAM'
+
+        saturation_model = SaturationModel()
+        saturation_model.data = np.zeros(shape=(2048, 2048), dtype=np.float32)
+        saturation_model.dq = np.zeros(shape=(2048, 2048), dtype=np.int32)
+        saturation_model.meta.subarray.xstart = 1
+        saturation_model.meta.subarray.ystart = 1
+        saturation_model.meta.subarray.xsize = 2048
+        saturation_model.meta.subarray.ysize = 2048
+        saturation_model.meta.instrument.name = 'NIRCAM'
+        saturation_model.meta.description = 'Fake data.'
+        saturation_model.meta.telescope = 'JWST'
+        saturation_model.meta.reftype = 'SaturationModel'
+        saturation_model.meta.author = 'Alicia'
+        saturation_model.meta.pedigree = 'Dummy'
+        saturation_model.meta.useafter = '2015-10-01T00:00:00'
+
+        return data_model, saturation_model
+
+    return _cube
+
+
+@pytest.fixture(scope='function')
+def setup_miri_cube():
+    ''' Set up fake MIRI data to test.'''
+
+    def _cube(xstart, ystart, ngroups, nrows, ncols):
+
+        nints = 1
+
+        # create the data and groupdq arrays
+        csize = (nints, ngroups, nrows, ncols)
+        data = np.full(csize, 1.0)
+        pixeldq = np.zeros((nrows, ncols), dtype=int)
+        groupdq = np.zeros(csize, dtype=int)
+
+        # create a JWST datamodel for MIRI data
+        data_model = MIRIRampModel(data=data, pixeldq=pixeldq, groupdq=groupdq)
```

Comment:
Same as my comment above for NIRCam.  The `data`, `pixeldq` and `groupdq` automatically get generated (filled with zeros of the correct `dtype`) when the constructor is passed a shape tuple.
```
data_model = MIRIRampModel(csize)
data_model.data += 1
```